### PR TITLE
JAMES-1756 Improve handling of error message when moving a message

### PR DESCRIPTION
**Pull request by Laura-Royet**

### CODE REVIEW

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java [hunk 1]
```diff
@@ -1408,4 +1410,46 @@ private boolean isHtmlMessageReceived(AccessToken recipientToken) {
...
+                .contentType(ContentType.JSON)
+                .header("Authorization", this.accessToken.serialize())
+                .body(requestBody)
+        .when()
+                .post("/jmap")
+        .then()
+                .statusCode(200)
+                .body(NAME, equalTo("messagesSet"))
+                .body(ARGUMENTS + ".notUpdated", hasKey(messageToMoveId))
+                .body(ARGUMENTS + ".notUpdated[\""+messageToMoveId+"\"].type", equalTo("invalidProperties"))
```
Comment (mbaechler) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java]: you should define a variable for `ARGUMENTS + ".notUpdated[\""+messageToMoveId+"\"]`

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java [hunk 2]
```diff
@@ -1408,4 +1410,46 @@ private boolean isHtmlMessageReceived(AccessToken recipientToken) {
...
+                .body(ARGUMENTS + ".notUpdated[\""+messageToMoveId+"\"].properties[0]", equalTo("mailboxIds"))
+                .body(ARGUMENTS + ".notUpdated[\""+messageToMoveId+"\"].description", equalTo("mailboxIds: Code is not implemented "
```
Comment (mbaechler) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java]: message is not good enough, should be "message move is not supported"

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java [hunk 3]
```diff
@@ -1408,4 +1410,46 @@ private boolean isHtmlMessageReceived(AccessToken recipientToken) {
             return false;
         }
     }
+    
+    @Test
+    public void movingAMessageIsNotSupported() throws Exception {
+        jmapServer.serverProbe().createMailbox("#private", username, "heartFolder");
+        Mailbox heartFolder = jmapServer.serverProbe().getMailbox("#private", username, "heartFolder");
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java]: Extract the chain "heartFolder" ?

Comment (Laura-Royet) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java]: done

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java (hunk 1 above)
Comment (Laura-Royet) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java]: done

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java (hunk 2 above)
Comment (Laura-Royet) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java]: done

**Comment by rouazana** [comment]
:+1:

**Comment by aduprat** [comment]
:+1:

**Pull request merged**